### PR TITLE
chore: upgrades collector to 0.29.0.

### PR DESCRIPTION
**Pull request by jcchavezs**
## Description

Upgrades OTel collector to 0.29.0.

### CODE REVIEW

File: cmd/collector/main.go [hunk 1]
```diff
@@ -23,14 +23,13 @@ func main() {
...
 
-	info := component.ApplicationStartInfo{
-		ExeName:  "collector",
-		LongName: "Hypertrace Collector",
-		Version:  Version,
-		GitHash:  GitHash,
+	info := component.BuildInfo{
+		Command:     "hypertrace-collector",
+		Description: "Hypertrace Collector",
+		Version:     Version,
```
Comment (davexroth) [inline cmd/collector/main.go:29]: curious why the GitHash was removed.

**Review by davexroth — APPROVED** [review]

File: cmd/collector/main.go [hunk 2]
```diff
@@ -23,14 +23,13 @@ func main() {
 		log.Fatalf("failed to build default components: %v", err)
 	}
 
-	info := component.ApplicationStartInfo{
-		ExeName:  "collector",
-		LongName: "Hypertrace Collector",
-		Version:  Version,
-		GitHash:  GitHash,
+	info := component.BuildInfo{
+		Command:     "hypertrace-collector",
```
Comment (davexroth) [inline cmd/collector/main.go]: this should still be collector right?  

If you do want to change the name of the executable, we'll need to update the Dockerfile too.

File: cmd/collector/main.go (hunk 1 above)
Comment (jcchavezs) [reply cmd/collector/main.go:29]: It is not supported anymore. We could check if version == dev then use dev-{hash} but not sure if that would be useful. Are you missing it for an specific reason?

File: processors/tenantidprocessor/tenantidprocessor_test.go [hunk 3]
```diff
@@ -173,7 +183,11 @@ func TestReceiveOTLPGRPC_Traces(t *testing.T) {
 		},
 	)
 	require.NoError(t, err)
-	reqTraces := GenerateTraceDataOneSpan()
+
+	err = tracesExporter.Start(context.Background(), componenttest.NewNopHost())
```
Comment (jcchavezs) [inline processors/tenantidprocessor/tenantidprocessor_test.go:187]: This was the change that was missing and was causing more pain. Amazingly this is a breaking change that is not explicit in the API but depends on the internal implementation.

**Pull request merged**